cmake/Runner.cmake

```diff
@@ -0,0 +1,38 @@
+# This source file is part of the Swift.org open source project
```

Comment:
Are we sure this file is necessary? The Swift open-source project uses SwiftPM to generate the relevant code here; I think the need for a separate runner may be specific to your use case, so should stay in your fork. (I _think_.)

Reply:
The entry point, as I understand it, is required to make a stand-alone test executable.  SwiftPM hides this from users, and my goal here is to make it so that someone building with CMake doesn't have to discover this fact and then have to figure out how to implement it themselves.  

Furthermore if the entry point signature ever changes, or additional scaffolding of whatever sort needs to be added, this would provide a central place for all projects to get something definitive automatically.  I'm happy to move it out to my project, but this is something every project would ultimately need, so I thought it made sense to provide it here.

Reply:
I'm uncertain too whether it belongs here. I'd suggest we break this piece out and consider it in a standalone PR following on from this one. It may be worthwhile, just seems less central to the initial goal. Also, the content of the generated file invokes an entry point function whose name specifically indicates it is meant for SwiftPM usage, so if we _do_ eventually include this, perhaps we should also generalize the name of that entry point and ensure it will behave well in other contexts.

Reply:
Let's remove this file from this PR. As Stuart says, we can consider it separately.

The `__swiftPMEntryPoint()` function exists solely for use by SwiftPM itself due to SwiftPM's unique constraints (being part of the Swift toolchain.) It has no guarantee of ABI/API stability. We don't want external build systems unintentionally relying on some aspect of this function signature that changes with Swift 6+_n_.

My colleague @dennisweissmann and I are working on defining a stable "ABI entry point" (see [here](https://github.com/search?q=repo%3Aapple%2Fswift-testing%20abientrypoint_v0&type=code)) that will, when complete, be a suitable substitute for non-SwiftPM tools. The implementation is identical to `__swiftPMEntryPoint()`, but the interface is meant to be stable over time.

That function is incompletely specified at this time, so for the purposes of the Browser Company's fork, I won't tell you not to use `__swiftPMEntryPoint()`. Just be prepared to switch to the ABI entry point. 😄

Reply:
Sounds good.

Reply:
(For historical context, I already had to mimic what SPM does for xctest, which involves finding all the test cases and test functions, and this was a lot easier.)